src/engine.rs

```diff
@@ -201,11 +220,20 @@ pub fn run(params: FunctionRunParams) -> Result<FunctionRunResult> {
 
     logs.extend_from_slice(error_logs.as_bytes());
 
+    // When codec is not specified, we infer it from the input
+    let inferered_codec = if uses_msgpack {
+        Codec::Messagepack
+    } else {
+        input.codec
+    };
```

Comment:
Hmm I'm a bit confused by this here, the output `BytesContainer` already does the right inferencing based on the input codec, mind explaining on why we need to infer it again?

Reply:
Right now, when I leave the `let input_stream = MemoryInputPipe::new(input.raw.clone());` the output for a wasm API is 
<img width="248" alt="Screenshot 2025-04-25 at 1 51 23 PM" src="https://github.com/user-attachments/assets/1a5a78d0-4356-4b07-a0b3-e7181a752d25" />

vs when setting the `--codec messagepack`, we get:

<img width="260" alt="Screenshot 2025-04-25 at 1 51 38 PM" src="https://github.com/user-attachments/assets/4949505b-59a3-48ad-9f14-a17d3fb57d4f" />

Not sure if this is the best way to fix it, but I think the input was not getting converted. 🤔